pkg/mcs/resource_manager/server/resource_group.go

```diff
@@ -167,7 +167,7 @@ func FromProtoResourceGroup(group *rmpb.ResourceGroup) *ResourceGroup {
 func (rg *ResourceGroup) RequestRU(now time.Time, neededTokens float64, targetPeriodMs uint64) *rmpb.GrantedRUTokenBucket {
 	rg.Lock()
 	defer rg.Unlock()
-	if rg.RUSettings == nil {
+	if rg.RUSettings == nil || rg.RUSettings.RU.Settings == nil {
```

Comment:
Should we check whether `rg.RUSettings.RU` is nil?

Reply:
`rg.RUSettings.RU` is not a pointer